filter.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"encoding/base64"
+	"strings"
+	"time"
+
+	"github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/api"
+)
+
+type filter struct {
+	callbacks api.FilterCallbackHandler
+	config    *config
+}
+
+const secretKey = "secret"
+
+// parseBasicAuth parses an HTTP Basic Authentication string.
+// "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==" returns ("Aladdin", "open sesame", true).
+func parseBasicAuth(auth string) (username, password string, ok bool) {
+	const prefix = "Basic "
+	// Case insensitive prefix match. See Issue 22736.
```

Comment:
Where is the issue 22736?

Reply:
Here's the issue https://github.com/golang/go/issues/22736  , I will make it clear

Reply:
Resolved.